test/core_test.exs

```diff
@@ -75,4 +95,10 @@ defmodule CoreTest do
     assert elem(result, 0) == :ok
     stop(:test_reporter)
   end
+
+  defp init_and_wait(metrics, opts) do
+    :ok = init(metrics, opts)
+    TelemetryMetricsPrometheus.Core.Registry.config(:test_reporter)
```

Comment:
This is a hack to ensure the `:setup` message has finished. This works since the config sends a message, so will be added to the message queue. I'll change it to use `get_state/1` instead.